test/test_toroid.py

```diff
@@ -0,0 +1,183 @@
+# A script to run test cases of toroid-ray.py
+
+
+# TODO: do this in a more pythonic way
+import os
+import sys
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+parent_dir = os.path.join(current_dir, "..")
+sys.path.insert(0, parent_dir)
+#
+
+import math
+import random
+import numpy as np
+import numpy.linalg as la
+import pytest
+import inspect
+import toroid
+from toroid import Toroid
+
+glob_rand_seed = 1999
+glob_rng = np.random.default_rng(seed=glob_rand_seed)
+
+# Quick shorthand to check if two arrays are equivalent
+def check_equal(a, b, rel_tol=1e-09, abs_tol=0.0):
+    assert len(a) == len(b)
+    for i in range(len(a)):
+        assert math.isclose(a[i], b[i], rel_tol=rel_tol, abs_tol=abs_tol)
+
+# In lieu of any rootfinder implementations, just use numpy for now
+def real_roots_numpy(coeffs: list[float]) -> (list[float], dict):
+    """
+    Solves for roots of the polynomial using numpy's eigenvalue / matrix implementation
+    """
+    all_roots = np.roots(coeffs)
+    real = all_roots[np.isreal(all_roots)]
+    real_list = [float(np.real(r)) for r in real]
+    return real_list, locals()
```

Comment:
Remove locals()

In general, don't include type names in variable names. So real_list could just be real_roots.

You can also simplify this a bit:

```
    roots = np.roots(coeffs)
    return [r.real for r in roots if np.isreal(r)]
```